soc/arm/microchip_mec/mec172x/soc.c

```diff
@@ -0,0 +1,417 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ * Copyright (c) 2021 Microchip Technology Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <sys/__assert.h>
+#include <device.h>
+#include <init.h>
+#include <soc.h>
+#include <kernel.h>
+#include <arch/cpu.h>
+#include <arch/arm/aarch32/cortex_m/cmsis.h>
+
+#define ECIA_XEC_REG_BASE						\
+	((struct ecia_ar_regs *)(DT_REG_ADDR(DT_NODELABEL(ecia))))
+#define ECS_XEC_REG_BASE						\
+	((struct ecs_regs *)(DT_REG_ADDR(DT_NODELABEL(ecs))))
+#define PCR_XEC_REG_BASE						\
+	((struct pcr_regs *)(DT_REG_ADDR(DT_NODELABEL(pcr))))
+#define VBATR_XEC_REG_BASE						\
+	((struct vbatr_regs *)(DT_REG_ADDR(DT_NODELABEL(vbr))))
+
+#define CLK32K_SIL_OSC_DELAY		256U
+#define CLK32K_PLL_LOCK_WAIT		(16U * 1024U)
+#define CLK32K_PIN_WAIT			4096U
+#define CLK32K_XTAL_WAIT		4096U
+#define CLK32K_XTAL_MON_WAIT		(64U * 1024U)
+
+#define DEST_PLL			0
+#define DEST_PERIPH			1
+
+#define CLK32K_FLAG_CRYSTAL_SE		BIT(0)
+#define CLK32K_FLAG_PIN_FB_CRYSTAL	BIT(1)
+
+#define PIN_32KHZ_IN_MODE (MCHP_GPIO_CTRL_PUD_NONE |\
+	MCHP_GPIO_CTRL_PWRG_VTR_IO | MCHP_GPIO_CTRL_IDET_DISABLE |\
+	MCHP_GPIO_CTRL_MUX_F1)
+
+enum clk32k_src {
+	CLK32K_SRC_SIL_OSC = 0,
+	CLK32K_SRC_CRYSTAL,
+	CLK32K_SRC_PIN,
+	CLK32K_SRC_MAX
+};
+
+enum clk32k_dest { CLK32K_DEST_PLL = 0, CLK32K_DEST_PERIPH, CLK32K_DEST_MAX };
+
+static uint32_t spin_delay(uint32_t cnt)
```

Comment:
add comment; what is this function doing?

Reply:
Hi @sjg20 Added comments